magda-sleuther-duplicate-records/src/onRecordFound.ts

```diff
@@ -0,0 +1,147 @@
+import * as _ from "lodash";
+import Registry from "@magda/typescript-common/dist/registry/AuthorizedRegistryClient";
+import { Record } from "@magda/typescript-common/dist/generated/registry/api";
+//import unionToThrowable from "@magda/typescript-common/dist/util/unionToThrowable";
+import { DuplicateRecordsAspect, DistURL } from "./duplicateRecordsAspectDef";
+import FTPHandler from "./FtpHandler";
+
+export var mochaObject: any = { isRunning: false };
+
+export var testGroups: DuplicateRecordsAspect[];
+
+export default async function onRecordFound(
+    record: Record,
+    registry: Registry,
+    retries: number = 5,
+    baseRetryDelaySeconds: number = 1,
+    base429RetryDelaySeconds = 60,
+    ftpHandler: FTPHandler = new FTPHandler()
+) {
+    const distributions: Record[] =
+        record.aspects["dataset-distributions"] &&
+        record.aspects["dataset-distributions"].distributions;
+
+    if (!distributions || distributions.length === 0) {
+        return Promise.resolve();
+    }
+
+    // DistContainer contains only the properties we need, putting all distributions into this will help manipulate the data.
+    const distributionContainers: DistContainer[] = _.flatten(
+        distributions.map(function(distribution) {
+            if (
+                distribution.aspects &&
+                distribution.aspects["dcat-distribution-strings"]
+            ) {
+                let distContainer: DistContainer[] = [];
+                distContainer.push({
+                    url: {
+                        url: null,
+                        type: null
+                    },
+                    id: null
+                });
+                distContainer.push({
+                    url: {
+                        url: null,
+                        type: null
+                    },
+                    id: null
+                });
+
+                
+                distContainer[0].url = {
+                    url: distribution.aspects["dcat-distribution-strings"]
+                        .accessURL as string,
+                    type: "accessURL" as "accessURL"
+                } || { type: "none" };
```

Comment:
I'm not sure exactly what this line is supposed to be doing - the `|| { type : "none" }` bit can never execute because the object before `||` is always going to be truthy. I'm guessing this is supposed to check whether there's an `accessURL` and set `{type: none}` if not?